source/speech.py

```diff
@@ -1939,3 +2008,489 @@ def __repr__(self):
 		if self.text:
 			out += ", text=%r" % self.text
 		return out + ")"
+
+class BaseCallbackCommand(SpeechCommand):
+	"""Base class for commands which cause a function to be called when speech reaches them.
+	This class should not be instantiated directly.
+	It is designed to be subclassed to provide specific functionality;
+	e.g. L{BeepCommand}.
+	To supply a generic function to run, use L{CallbackCommand}.
+	This command is never passed to synth drivers.
+	"""
+
+	def run(self):
+		"""Code to run when speech reaches this command.

```

Comment:
Could you add something to the doc string that explains how this method is executed? As far as I understand, these methods block the main thread as long as they're running. That's pretty important information IMO.